--- SDK.UBNT.v3.4.4416.orig/conf/xs2/busybox.config	2008-10-20 23:34:04.000000000 +0200
+++ SDK.UBNT.v3.4.4416/conf/xs2/busybox.config	2009-11-19 16:20:29.000000000 +0100
@@ -67,7 +67,7 @@
 #
 # Coreutils
 #
-# CONFIG_BASENAME is not set
+CONFIG_BASENAME=y
 # CONFIG_CAL is not set
 CONFIG_CAT=y
 # CONFIG_CHGRP is not set
@@ -81,7 +81,7 @@
 # CONFIG_FEATURE_DATE_ISOFMT is not set
 CONFIG_DD=y
 # CONFIG_DF is not set
-# CONFIG_DIRNAME is not set
+CONFIG_DIRNAME=y
 # CONFIG_DOS2UNIX is not set
 # CONFIG_DU is not set
 CONFIG_ECHO=y
@@ -90,7 +90,8 @@
 # CONFIG_EXPR is not set
 CONFIG_FALSE=y
 # CONFIG_FOLD is not set
-# CONFIG_HEAD is not set
+CONFIG_HEAD=y
+CONFIG_FEATURE_FANCY_HEAD=n
 # CONFIG_HOSTID is not set
 # CONFIG_ID is not set
 # CONFIG_INSTALL is not set
@@ -134,7 +135,7 @@
 #
 # CONFIG_FEATURE_TEST_64 is not set
 CONFIG_TOUCH=y
-# CONFIG_TR is not set
+CONFIG_TR=y
 CONFIG_TRUE=y
 # CONFIG_TTY is not set
 # CONFIG_UNAME is not set
--- SDK.UBNT.v3.4.4416.orig/conf/xs2/product.mk	2009-05-21 14:03:36.000000000 +0200
+++ SDK.UBNT.v3.4.4416/conf/xs2/product.mk	2009-11-19 14:06:10.000000000 +0100
@@ -35,7 +35,7 @@
 APPS_GPL := matrixssl php2 boa busybox udhcp dnsmasq scanparser \
 	    ppp tinysnmp iptables ebtables \
 	    wireless-tools hostapd-0.6.3 wpasupplicant-0.6.3 \
-	    iproute2  trigger_url ntpclient igmpproxy dropbear
+	    iproute2  trigger_url ntpclient igmpproxy dropbear coova-chilli haserl
 #	    bird batmand wifidog
 #APPS_GPL += libpcap tcpdump iperf
 
--- SDK.UBNT.v3.4.4416.orig/conf/xs2/linux-kernel-2.4.config	2008-12-24 13:54:30.000000000 +0100
+++ SDK.UBNT.v3.4.4416/conf/xs2/linux-kernel-2.4.config	2009-11-19 15:52:37.000000000 +0100
@@ -525,7 +525,7 @@
 # CONFIG_DUMMY is not set
 # CONFIG_BONDING is not set
 # CONFIG_EQUALIZER is not set
-# CONFIG_TUN is not set
+CONFIG_TUN=m
 # CONFIG_ETHERTAP is not set
 
 #
--- SDK.UBNT.v3.4.4416.orig/rootfs/ls2/linuxrc	2009-04-17 12:55:29.000000000 +0200
+++ SDK.UBNT.v3.4.4416/rootfs/ls2/linuxrc	2009-11-19 14:29:47.000000000 +0100
@@ -74,7 +74,7 @@
 # making sure that critical files are in place
 mkdir -p /etc/rc.d /etc/init.d
 # forced update
-for f in inittab rc.d/rc.sysinit rc.d/rc rc.d/rc.stop ppp; do
+for f in inittab rc.d/rc.sysinit rc.d/rc rc.d/rc.stop ppp chilli chilli.conf; do
 	cp -f -r /usr/etc/$f /etc/$f
 done
 
--- SDK.UBNT.v3.4.4416.orig/rootfs/ls2/usr/etc/rc.d/rc.sysinit	2008-06-06 13:27:08.000000000 +0200
+++ SDK.UBNT.v3.4.4416/rootfs/ls2/usr/etc/rc.d/rc.sysinit	2009-11-19 14:31:40.000000000 +0100
@@ -17,3 +17,8 @@
 if [ -f /etc/persistent/rc.postsysinit ]; then
        	. /etc/persistent/rc.postsysinit
 fi
+
+if [ -f /etc/chilli/rc ]; then
+       	. /etc/chilli/rc
+fi
+
